Automattic/Gravatar-SDK-Android
# Initial implementation of the QuickEditor with the provided token

Closes #251

### Description

**This is a DRAFT PR for now to discuss this solution**

An initial implementation of the `GravatarQuickEditor` that can be launched with the given auth token. This changes the flow and skips the OAuth completely since the third-party app should already be authorized to perform actions.

Because this is an "outside" token that is handled by the third-party app developers I don't want to store and only keep it in the "in-memory" token storage. For that I created an interface `TokenStorage` and now we will have two: DataStore and In-Memory implementations. A proper implementation will be injected to `AvatarsRepository` depending on how the QE was launched.

The above solution was mainly driven by the fact that `AvatarPicker` has no information about the token and I wanted to keep it that way. 

What do you think about that? Do you have other ideas how to handle this? 

### Testing Steps

You can add the token to the `local.properties` file like that:
`demo-app.bearer.token={token}`

This will be read in the demo app. Tap the checkbox next to the token if you want to launch the QE with it. 
<img width="394" alt="Screenshot 2024-08-23 at 12 52 06" src="https://github.com/user-attachments/assets/87c5f42e-4b28-4edf-91ba-5b4317ef3561">


Testing the Log in/Close CTA:

1. With the provided token, it's simple - just use invalid token.
2. For the OAuth flow to ensure there's no regression use Charles of another app to mock network response.
